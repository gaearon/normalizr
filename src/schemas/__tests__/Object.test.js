// eslint-env jest
import { fromJS } from 'immutable';
import { denormalize, schema } from '../..';

describe(`${schema.Object.name} denormalization`, () => {
  test('denormalizes plain object shorthand with missing schemas', () => {
    const userSchema = new schema.Entity('user');
    const entities = {
      user: {
        1: { id: 1, name: 'Jane' },
      },
    };
    expect(denormalize({ user: 1 }, { user: userSchema, tacos: {} }, entities)).toMatchSnapshot();
    expect(denormalize({ user: 1 }, { user: userSchema, tacos: {} }, fromJS(entities))).toMatchSnapshot();
    expect(denormalize(fromJS({ user: 1 }), { user: userSchema, tacos: {} }, fromJS(entities))).toMatchSnapshot();
  });
});
