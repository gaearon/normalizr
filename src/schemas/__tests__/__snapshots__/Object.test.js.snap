// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectSchema denormalization denormalizes plain object shorthand with missing schemas 1`] = `
Object {
  "user": Object {
    "id": 1,
    "name": "Jane",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand with missing schemas 2`] = `
Object {
  "user": Immutable.Map {
    "id": 1,
    "name": "Jane",
  },
}
`;

exports[`ObjectSchema denormalization denormalizes plain object shorthand with missing schemas 3`] = `
Immutable.Map {
  "user": Immutable.Map {
    "id": 1,
    "name": "Jane",
  },
}
`;
